modules/crop-workflowoperation/src/main/java/org/opencastproject/workflow/handler/crop/CropWorkflowOperationHandler.java

```diff
@@ -0,0 +1,189 @@
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * The workflow definition will run recordings to crop them from 4:3 to 16:9.
+ */
+public class CropWorkflowOperationHandler extends AbstractWorkflowOperationHandler {
+
+  /** The logging facility */
+  private static final Logger logger = LoggerFactory.getLogger(CropWorkflowOperationHandler.class);
+
+  /** Name of the configuration key that specifies the flavor of the track to analyze */
+  private static final String PROP_ANALYSIS_TRACK_FLAVOR = "source-flavor";
+
+  /** Name of the configuration key that specifies the flavor of the track to analyze */
+  private static final String PROP_TARGET_TAGS = "target-tags";
+
+  /** The configuration options for this handler */
+  private static final SortedMap<String, String> CONFIG_OPTIONS;
+
+  /** Id builder used to create ids for cropped tracks */
+  private final IdBuilder idBuilder = IdBuilderFactory.newInstance().newIdBuilder();
+
+  static {
+    CONFIG_OPTIONS = new TreeMap<>();
+    CONFIG_OPTIONS.put(PROP_ANALYSIS_TRACK_FLAVOR,
+            "The flavor of the track to analyze. If multiple tracks match this flavor, the first will be used.");
+    CONFIG_OPTIONS.put(PROP_TARGET_TAGS, "The tags to apply to the resulting mpeg-7 catalog");
+  }
+
+  /** The composer service */
+  private CropService cropService = null;
+
+  /** The local workspace */
+  private Workspace workspace = null;
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see WorkflowOperationHandler#getConfigurationOptions()
+   */
+  @Override
+  public SortedMap<String, String> getConfigurationOptions() {
+    return CONFIG_OPTIONS;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.workflow.api.WorkflowOperationHandler#start(org.opencastproject.workflow.api.WorkflowInstance, JobContext)
+   */
+  @Override
+  public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobContext jobContext)
+          throws WorkflowOperationException {
+    logger.debug("Running cropping on workflow {}", workflowInstance.getId());
+
+    final String targetTrackId = idBuilder.createNew().toString();
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+    MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+
+    String trackFlavor = StringUtils.trimToNull(operation.getConfiguration(PROP_ANALYSIS_TRACK_FLAVOR));
+    List<String> targetTags = asList(operation.getConfiguration(PROP_TARGET_TAGS));
+    List<Track> candidates = new ArrayList<>();
+    if (trackFlavor != null) {
+      candidates.addAll(Arrays.asList(mediaPackage.getTracks(MediaPackageElementFlavor.parseFlavor(trackFlavor))));
+    } else {
+      candidates.addAll(Arrays.asList(mediaPackage.getTracks(MediaPackageElements.PRESENTATION_SOURCE)));
+    }
+
+    Iterator<Track> ti = candidates.iterator();
```

Comment:
Using `…stream().filter(…)…` is probably easier.

Reply:
I was getting a ArrayIndexOutOfBoundsException when building the module